File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/documents/IsCaseDocument.scala
package dk.openesdh.repo.webscripts.documents

import com.typesafe.scalalogging.slf4j.StrictLogging
import dk.openesdh.repo.model.OpenESDHModel
import dk.openesdh.repo.services.cases.CaseService
import dk.openesdh.repo.services.documents.DocumentService
import org.alfresco.service.cmr.repository.{NodeService, InvalidNodeRefException, NodeRef}
import org.apache.commons.lang.StringUtils
import org.json.JSONObject
import org.springframework.extensions.webscripts._

/**
  * @author Lanre Abiwon
  */
class IsCaseDocument( val documentService: DocumentService, val nodeService: NodeService, val caseService: CaseService) extends AbstractWebScript with StrictLogging{

    protected override def execute(req: WebScriptRequest, resp: WebScriptResponse) = {
      val templateArgs = req.getServiceMatch.getTemplateVars

      try {
          val storeType: String = templateArgs.get ("store_type")
          val storeId: String = templateArgs.get ("store_id")
          val id: String = templateArgs.get ("id")
          val docNodeRefStr = s"$storeType://$storeId/$id"
          val documentNode: NodeRef = new NodeRef (docNodeRefStr)
          val caseNodeRef = documentService.getCaseNodeRef(documentNode)
          val caseId =  caseService.getCaseId(caseNodeRef)
          val obj: JSONObject = new JSONObject()

          if (caseNodeRef != null ){
            obj.put("isCaseDoc", true)
            obj.put("caseId", caseId)
          }
          else
            obj.put("isCaseDoc", false)

          resp.getWriter.write(obj.toString)
      }
      catch {
        case inre: InvalidNodeRefException => {
          //logger.error("", inre)
        }
      }
    }

 }

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/documents/IsCaseDocument.java
package dk.openesdh.repo.webscripts.documents;

import java.io.IOException;
import java.util.Map;

import org.alfresco.service.cmr.repository.InvalidNodeRefException;
import org.alfresco.service.cmr.repository.NodeRef;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.json.JSONException;
import org.json.JSONObject;
import org.springframework.extensions.webscripts.AbstractWebScript;
import org.springframework.extensions.webscripts.WebScriptRequest;
import org.springframework.extensions.webscripts.WebScriptResponse;

import dk.openesdh.repo.services.cases.CaseService;
import dk.openesdh.repo.services.documents.DocumentService;

/**
 * @Author Lanre
 */
public class IsCaseDocument extends AbstractWebScript {

    private static final Log logger = LogFactory.getLog(DocumentCaseContainers.class);
    private CaseService caseService;
    private DocumentService documentService;

    @Override
    public void execute(WebScriptRequest req, WebScriptResponse res) throws IOException {
        Map<String, String> templateArgs = req.getServiceMatch().getTemplateVars();
        String storeType = templateArgs.get("store_type");
        String storeId = templateArgs.get("store_id");
        String id = templateArgs.get("id");
        String docNodeRefStr = storeType + "://" + storeId + "/" + id;
        NodeRef documentNode = new NodeRef(docNodeRefStr);

        JSONObject json = new JSONObject();
        try {

            NodeRef caseNodeRef = documentService.getCaseNodeRef(documentNode);
            String caseId = caseService.getCaseId(caseNodeRef);

            if (caseNodeRef != null) {
                json.put("isCaseDoc", true);
                json.put("caseId", caseId);
                json.write(res.getWriter());
            } else {
                json.put("isCaseDoc", false);
                json.write(res.getWriter());
            }
        } catch (InvalidNodeRefException inre) {
            logger.error("The invalid nodeRef exception: " + inre.getMessage());
        } catch (JSONException jse) {
            logger.error("Unable to build teh json model because of the following exception: " + jse.getMessage());
        }

    }

    public void setCaseService(CaseService caseService) {
        this.caseService = caseService;
    }

    public void setDocumentService(DocumentService documentService) {
        this.documentService = documentService;
    }

}
